osu.Framework/Graphics/UserInterface/TextBox.cs

```diff
@@ -450,6 +451,10 @@ private bool removeCharacterOrSelection(bool sound = true)
 
             text = text.Remove(start, count);
 
+            // Reorder characters depth after removal to avoid ordering issues with newly added characters.
+            for (int i = 0; i < TextFlow.Count; i++)
```

Comment:
I think
```suggestion
            for (int i = start; i < TextFlow.Count; i++)
```
is a valid optimisation to make here, since the first `start` characters should have their depth set appropriately. Seems to pass tests, at least. It's not intensely important though; maybe even better left as-is to be absolutely sure of correctness.

Reply:
Would need input on this one, I'm aware of this causing the bug to show up again somehow and causing a twisted issue to solve in the future.

Reply:
If it's passing your tests and you're still cautious of it, this means your tests are not covering the issue.